File: app/src/main/java/com/tematikhonov/androidgeekbrainssecondapp/MainActivity.java
package com.tematikhonov.androidgeekbrainssecondapp;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBarDrawerToggle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.drawerlayout.widget.DrawerLayout;

import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;
import android.widget.Toast;

import com.google.android.material.navigation.NavigationView;
import com.tematikhonov.androidgeekbrainssecondapp.data.Navigation;
import com.tematikhonov.androidgeekbrainssecondapp.data.Publisher;
import com.tematikhonov.androidgeekbrainssecondapp.ui.NotesListFragment;
import com.tematikhonov.androidgeekbrainssecondapp.ui.StartFragment;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "myLogs";
    private Navigation navigation;
    private Publisher publisher = new Publisher();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        navigation = new Navigation(getSupportFragmentManager());
        initView();
        getNavigation().addFragment(StartFragment.newInstance(), false);

    }

    private void initView() {
        Toolbar toolbar = initToolbar();
        initDrawer(toolbar);
    }

    private void initDrawer(Toolbar toolbar) {
        final DrawerLayout drawer = findViewById(R.id.drawer_layout);
        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
                this, drawer, toolbar,
                R.string.navigation_drawer_open,
                R.string.navigation_drawer_close);
        drawer.addDrawerListener(toggle);
        toggle.syncState();
        NavigationView navigationView = findViewById(R.id.nav_view);
        navigationView.setNavigationItemSelectedListener(new NavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                int id = item.getItemId();
                switch (id) {
                    case R.id.action_main:
                        Toast.makeText(getBaseContext(), "Main", Toast.LENGTH_LONG).show();
                        return true;
                    case R.id.action_settings:
                        Intent intent = new Intent(MainActivity.this, SettingsActivity.class);
                        startActivity(intent);
                        return true;
                    case R.id.action_favorite:
                        Toast.makeText(getBaseContext(), "Favorite", Toast.LENGTH_LONG).show();
                        return true;
                    case R.id.action_journal:
                        Toast.makeText(getBaseContext(), "Journal", Toast.LENGTH_LONG).show();
                        return true;
                    case R.id.action_about:
                        Toast.makeText(getBaseContext(), "About", Toast.LENGTH_LONG).show();
                        return true;
                }
                return false;
            }
        });
    }

    private Toolbar initToolbar() {
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setHomeButtonEnabled(true);
        return toolbar;
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }

    public Navigation getNavigation() {
        return navigation;
    }

    public Publisher getPublisher() {
        return publisher;
    }
}

File: app/src/main/java/com/tematikhonov/androidgeekbrainssecondapp/MainActivity.kt
package com.tematikhonov.androidgeekbrainssecondapp

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.ActionBarDrawerToggle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.drawerlayout.widget.DrawerLayout
import com.google.android.material.navigation.NavigationView
import com.tematikhonov.androidgeekbrainssecondapp.data.*
import com.tematikhonov.androidgeekbrainssecondapp.ui.NotesListFragment

class MainActivity : AppCompatActivity() {
    var navigation: Navigation? = null
        private set
    val publisher = Publisher()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        navigation = Navigation(supportFragmentManager)
        initView()

    }

    private fun initView() {
        val toolbar = initToolbar()
        initDrawer(toolbar)
        navigation?.addFragment(NotesListFragment(), false)

    }

    private fun initDrawer(toolbar: Toolbar) {
        val drawer = findViewById<DrawerLayout>(R.id.drawer_layout)
        val toggle = ActionBarDrawerToggle(
                this, drawer, toolbar,
                R.string.navigation_drawer_open,
                R.string.navigation_drawer_close)
        drawer.addDrawerListener(toggle)
        toggle.syncState()
        val navigationView = findViewById<NavigationView>(R.id.nav_view)
        navigationView.setNavigationItemSelectedListener(NavigationView.OnNavigationItemSelectedListener { item ->
            val id = item.itemId
            when (id) {
                R.id.action_main -> {
                    Toast.makeText(baseContext, "Main", Toast.LENGTH_LONG).show()
                    return@OnNavigationItemSelectedListener true
                }
                R.id.action_settings -> {
                    val intent = Intent(this@MainActivity, SettingsActivity::class.java)
                    startActivity(intent)
                    return@OnNavigationItemSelectedListener true
                }
                R.id.action_favorite -> {
                    Toast.makeText(baseContext, "Favorite", Toast.LENGTH_LONG).show()
                    return@OnNavigationItemSelectedListener true
                }
                R.id.action_journal -> {
                    Toast.makeText(baseContext, "Journal", Toast.LENGTH_LONG).show()
                    return@OnNavigationItemSelectedListener true
                }
                R.id.action_about -> {
                    Toast.makeText(baseContext, "About", Toast.LENGTH_LONG).show()
                    return@OnNavigationItemSelectedListener true
                }
            }
            false
        })
    }

    private fun initToolbar(): Toolbar {
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.setHomeButtonEnabled(true)
        return toolbar
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }



    companion object {
        private const val TAG = "myLogs"
    }
}